Source/Particles/Collision/CollisionType.cpp

```diff
@@ -0,0 +1,233 @@
+    int coord_sys;
+    pp_check.query("coord_sys",coord_sys);
+    if (coord_sys != 0)
+    { amrex::Abort("Collision only works for Cartesian coordinate for now."); }
+
+    // check the dimension is 3
+    std::vector<std::string> prob_lo;
+    pp_check.getarr("prob_lo", prob_lo);
+    if (prob_lo.size() != 3)
+    { amrex::Abort("Collision only works for 3D for now."); }
+
+    // read collision species
+    std::vector<std::string> collision_species;
+    amrex::ParmParse pp(collision_name);
+    pp.getarr("species", collision_species);
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(collision_species.size() == 2,
+    "Collision species must name exactly two species.");
+
+    // default Coulomb log, if < 0, will be computed automatically
+    m_CoulombLog = -1.0;
+    pp.query("CoulombLog", m_CoulombLog);
+
+    for (int i=0; i<species_names.size(); i++)
+    {
+        if (species_names[i] == collision_species[0])
+        { m_species1_index = i; }
+        if (species_names[i] == collision_species[1])
+        { m_species2_index = i; }
+    }
+
+    if (collision_species[0] == collision_species[1])
+        m_isSameSpecies = true;
+    else
+        m_isSameSpecies = false;
+
+}
+
+using namespace amrex;
+// Define shortcuts for frequently-used type names
+using ParticleType = WarpXParticleContainer::ParticleType;
+using ParticleTileType = WarpXParticleContainer::ParticleTileType;
+using ParticleBins = DenseBins<ParticleType>;
+using index_type = ParticleBins::index_type;
+
+namespace {
+
+    /* Find the particles and count the particles that are in each cell.
+       Note that this does *not* rearrange particle arrays */
+    ParticleBins
+    findParticlesInEachCell( int const lev, MFIter const& mfi,
+                             ParticleTileType const& ptile) {
+
+        // Extract particle structures for this tile
+        int const np = ptile.numParticles();
+        ParticleType const* particle_ptr = ptile.GetArrayOfStructs()().data();
+
+        // Extract box properties
+        Geometry const& geom = WarpX::GetInstance().Geom(lev);
+        Box const& cbx = mfi.tilebox(IntVect::TheZeroVector()); //Cell-centered box
+        const auto lo = lbound(cbx);
+        const auto dxi = geom.InvCellSizeArray();
+        const auto plo = geom.ProbLoArray();
+
+        // Find particles that are in each cell ;
+        // results are stored in the object `bins`.
+        ParticleBins bins;
+        bins.build(np, particle_ptr, cbx,
+            // Pass lambda function that returns the cell index
+            [=] AMREX_GPU_HOST_DEVICE (const ParticleType& p) noexcept -> IntVect
+            {
+                return IntVect(AMREX_D_DECL((p.pos(0)-plo[0])*dxi[0] - lo.x,
+                                            (p.pos(1)-plo[1])*dxi[1] - lo.y,
+                                            (p.pos(2)-plo[2])*dxi[2] - lo.z));
+            });
+
+        return bins;
+    }
+
+}
+
+
+/* \brief Perform Coulomb collisions within one particle tile */
```

Comment:
Could you re-format this doc-string for Doxygen? e.g.,
```c++
/**
 * \brief Perform Coulomb collisions within one particle tile
 */
```
(and you may comment some parameters)